azimuth/modules/dataset_analysis/dataset_warnings.py

```diff
@@ -193,135 +195,117 @@ def get_general_warnings(
+            df = ds.remove_columns(
+                [
+                    column
+                    for column in ds.column_names
+                    if column not in [DatasetColumn.word_count, self.config.columns.label]
+                ]
+            ).to_pandas()
+
+            value_per_agg_per_split[split][Agg.mean] = df["word_count"].mean()
+            value_per_agg_per_split[split][Agg.std] = df["word_count"].std()
+
+            # Get mean and std word count per label
+            stats_per_label = df.groupby("label").agg(["mean", "std"]).reindex(class_index)
+            for agg in [Agg.mean, Agg.std]:
+                value_per_label_per_agg_per_split[split][agg] = stats_per_label["word_count"][
+                    agg
+                ].to_numpy()
+
+            # Get word count histogram per label. Columns are word counts and rows are labels.
+            hist_per_label_per_split[split] = (
+                df.groupby(["label", "word_count"])
+                .size()
+                .unstack(fill_value=0)
+                .reindex(class_index, fill_value=0)
+            )
 
-        # Get counts, this will have N columns with count and LABEL rows.
-        train_count = (
-            train_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
-        )
-        eval_count = (
-            eval_df.groupby(["label", DatasetColumn.word_count]).size().unstack(fill_value=0)
+        # Compute divergence and alerts
+        divergence_per_label_per_agg = dict()
+        alert_per_label_per_agg = dict()
+        thresholds_per_agg = {
+            Agg.mean: self.config.dataset_warnings.max_delta_mean_words,
+            Agg.std: self.config.dataset_warnings.max_delta_std_words,
+        }
+        for agg, threshold in thresholds_per_agg.items():
+            divergence_per_label_per_agg[agg] = np.abs(
+                value_per_label_per_agg_per_split[DatasetSplitName.train][agg]
+                - value_per_label_per_agg_per_split[DatasetSplitName.eval][agg]
+            )
+            alert_per_label_per_agg[agg] = divergence_per_label_per_agg[agg] > threshold
+        alert_per_label = np.any(
+            [alert_per_label_per_agg[Agg.mean], alert_per_label_per_agg[Agg.std]], axis=0
         )
 
-        train_missing_cols = find_missing_rows(train_count, cls_index)
-        eval_missing_cols = find_missing_rows(eval_count, cls_index)
-        train_count = fill_at_missing_indices(train_count, train_missing_cols)
-        eval_count = fill_at_missing_indices(eval_count, eval_missing_cols)
-        # Add to mean too.
-        train_desc = fill_at_missing_indices(
-            train_desc, train_missing_cols, fill_value=float("nan")
+        # Fill columns so both splits have the same values from 1 to N.
+        N = max(
+            [
+                max(hist_per_label_per_split[split].columns)
+                for split in self.available_dataset_splits
+            ]
```

Comment:
You shouldn't need the list comprehension here, `max()` accepts a generator directly.
```suggestion
            max(hist_per_label_per_split[split].columns) for split in self.available_dataset_splits
```

Reply:
Oh awesome!

Reply:
I fixed 3 others like that.